data/src/test/java/org/apache/iceberg/data/DataTestHelpers.java

```diff
@@ -43,7 +42,9 @@ public static void assertEquals(Types.StructType struct, Record expected, Record
   public static void assertEquals(Types.ListType list, List<?> expected, List<?> actual) {
     Type elementType = list.elementType();
 
-    Assert.assertEquals("List size should match", expected.size(), actual.size());
+    Assertions.assertThat(actual)
+        .hasSameSizeAs(expected)
+        .describedAs("List size should match");
```

Comment:
Instead of `describedAs` this should be using `.as()`: `assertThat(...).as(...).isEqualTo(...)`